homeassistant/components/homewizard/sensor.py

```diff
@@ -376,18 +369,135 @@ class HomeWizardSensorEntityDescription(
+        )
+    ) and coordinator.data.data.gas_unique_id is not None:
+
+        ent_reg.async_update_entity(
+            entity_id,
+            new_unique_id=f"{DOMAIN}_{coordinator.data.data.gas_unique_id}",
+        )
+
+    # Initialize default sensors
+    entities: list = [
         HomeWizardSensorEntity(coordinator, entry, description)
         for description in SENSORS
         if description.value_fn(coordinator.data.data) is not None
-    )
+    ]
+
+    # Initialize external devices
+    if coordinator.data.data.external_devices is not None:
+        for (unique_id, device) in coordinator.data.data.external_devices.items():
+
+            if description := EXTERNAL_SENSORS.get(device.meter_type):
+                entities.append(
+                    HomeWizardExternalSensorEntity(coordinator, description, unique_id)
+                )
+
+    async_add_entities(entities)
+
+
+class HomeWizardExternalSensorEntity(HomeWizardEntity, SensorEntity):
+    """Representation of externally connected HomeWizard Sensor."""
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        description: SensorEntityDescription,
+        device_unique_id: str,
+    ) -> None:
+        """Initialize Externally connected HomeWizard Sensors."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{DOMAIN}_{device_unique_id}"
+        self._device_id = device_unique_id
+        self.entity_description = description
+
+        if self._attr_device_info is not None:
+            self._attr_device_info[ATTR_IDENTIFIERS] = {(DOMAIN, self._attr_unique_id)}
+            self._attr_device_info[
+                ATTR_NAME
+            ] = f"{self.entity_description.key} ({self._device_id})"
+
+            if self.coordinator.data.device.serial is not None:
+                self._attr_device_info[ATTR_VIA_DEVICE] = (
+                    DOMAIN,
+                    self.coordinator.data.device.serial,
+                )
+
+    @property
+    def native_value(self) -> float | int | str | None:
+        """Return the sensor value."""
+        return self.device.value if self.device is not None else None
+
+    @property
+    def device(self) -> ExternalDevice | None:
+        """Return ExternalDevice object."""
+        return (
+            self.coordinator.data.data.external_devices[self._device_id]
+            if self.coordinator.data.data.external_devices is not None
+            else None
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return availability of meter."""
+        return super().available and self.native_value is not None
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return unit of measurement based on device unit."""
+        if (device := self.device) is None:
+            return None
+
+        if device.unit == "m3":
+            return UnitOfVolume.CUBIC_METERS
+
+        return device.unit
```

Comment:
I do not like this but I can't think of another way.

The unit is not predefined and can be everything, but in practice we see only 'm3' and 'GJ' for external devices.

Another problem is that is is theoretically possible to have an invalid unit for the device class. (`Gas` can have 'GJ' or 'Cats' for example.)

Reply:
We also can return the most likely unit for the device type, but that can trigger some issues when the unit is unexpected (which is also possible with this implementation). 

E.g. If we return GJ for 'Energy' and a device decided to use 'kWh' as unit.